xc7/fasm2bels/connection_db_utils.py

```diff
@@ -14,6 +33,22 @@ def get_tile_type_pkey(tile):
 
     @functools.lru_cache(maxsize=None)
     def maybe_get_wire(tile, wire):
```

Comment:
This seems weird.  What is generating the wrong wire names?

Reply:
With the https://github.com/SymbiFlow/symbiflow-arch-defs/pull/1242 some wire renaming was introduced. It is done during graph edge annotation with fasm features. The thing with `IOI_SING` tiles is that the "top" and "bottom" SING tile of a column uses different subsets of IOI tile bits and that is why it is needed. Moreover, some wires equivalent to those in "normal" IOIs have "SING" in their names. But since only the fasm features are renamed, not wires in the database, in order to successfully decode `IOI_SING` routing those wires need to be renamed back.

In order to avoid all of that we would probably need to create different tile types for "top" and "bottom" SINGs.

Reply:
So the bad names are coming from the FASM features?  Let's fix this at the FASM level then, rather than here.

Reply:
Ok, that makes sense. Done.